libntxm/arm7/source/player.cpp

```diff
@@ -273,8 +371,21 @@ void Player::stopChannel(u8 channel)
 	}
 }
 
+
 void Player::playTimerHandler(void)
 {
+	// cursor calculations, may as well reuse the same timer
+	if (state.playing_piano_sample)
+	{
+		u32 sample_passed_time = getTicks() - sample_lastms;
+		if (sample_passed_time >= 16) // only calculate roughly every frame. (1 frame = 16.6666...ms)
```

Comment:
note: this is not the cause of the drifting as it is still the exact same when calculated 1000 times per sec